### PR TITLE
added legend changes - colour should be replaced

### PR DESCRIPTION
**Pull request by DanielNowak98**
I have added style changes for the legend. However, due to the color alteration, the labels of the individual scopes are hardly readable. Therefore, I have temporarily made them white. The color can still be changed.
Furthermore, I have changed the location of the legend. It is now positioned in the top right corner, reducing the risk of covering parts of the plot.

added to `Style.cpp`: 

```
void SetImPlotLegendProperties() {
  // Legenden-Hintergrundfarbe ändern
  ImPlotStyle &plotStyle = ImPlot::GetStyle();
  plotStyle.Colors[ImPlotCol_LegendBg] =
      ImVec4(0.5f, 0.5f, 0.5f, 1.0f); // Dunkelgrau
}

```

and to `Style.hpp`: 
`void SetImPlotLegendProperties();`

and this to main: 

```
    SetImPlotLegendProperties();

    addPlots("Recording the data", captureData,
             [&sampler, &xmax_paused](auto /*x_min*/, auto x_max) {
               if (!flagPaused) {
                 ImPlot::SetupAxes("x [Data points]", "y [ADC Value]",
                                   ImPlotAxisFlags_AutoFit,
                                   ImPlotAxisFlags_AutoFit);
                 ImPlot::SetupAxisLimits(ImAxis_X1, x_max - 7500, x_max + 7500,
                                         ImGuiCond_Always);
               } else {
                 xmax_paused = x_max;
                 ImPlot::SetupAxes("x [Seconds]", "y [Volts]");
                 // x and y axes ranges: [0, 10], [-10, 200]
                 ImPlot::SetupAxesLimits(0, 10, -10, 200);
                 // make specific values/ticks on Y-axis
                 ImPlot::SetupAxisTicks(ImAxis_Y1, -10, 200, 22, nullptr, true);
                 ImPlot::SetupLegend(ImPlotLocation_NorthEast);
               }
             });

    ImGui::EndChild();
```

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -86,7 +86,7 @@ int main() {
     update_language_from_github();
   }
 
-  constexpr ImVec2 toolBtnSize{80, 80}; // toolbar buttons size
+  constexpr ImVec2 toolBtnSize = ImVec2(80, 80); // toolbar buttons size
```
Comment (NP199) [inline src/main.cpp:89]: Why was this line changed?

File: src/main.cpp [hunk 2]
```diff
@@ -188,10 +192,10 @@ int main() {
     load_settings(config);
     Style::SetupImGuiStyle(false, 0.99f);
     ImGui::SetNextWindowPos(ImVec2(0.0f, 0.0f));
-    auto windowSize{ImGui::GetIO().DisplaySize};
-    ImGui::SetNextWindowSize(windowSize);
+    ImGui::SetNextWindowSize(ImGui::GetIO().DisplaySize);
```
Comment (NP199) [inline src/main.cpp:195]: Here the window size variable is deleted?

Comment (DanielNowak98) [reply src/main.cpp:195]: this should not be deleted... idk why

File: src/main.cpp [hunk 3]
```diff
@@ -117,20 +117,22 @@ int main() {
...
+  auto addPlots = [&, firstRun = std::set<std::string>{}](
+                      auto const &name, auto const &plots,
+                      auto axesSetup) mutable {
     auto const plotRegion = ImGui::GetContentRegionAvail();
     if (ImPlot::BeginPlot(name, plotRegion)) {
       double x_min = std::numeric_limits<double>::max();
       double x_max = std::numeric_limits<double>::min();
 
-      for (auto const &plot : plots)
+      for (auto const &plot : plots) {
```
Comment (R-Abbasi) [inline src/main.cpp:128]: This is not required. Some people, however, find adding an extra curly bracket like that helpful in terms of readability but it usually also adds one line to the existing code which can decrease readability.

File: src/main.cpp [hunk 4]
```diff
@@ -117,20 +117,22 @@ int main() {
...
         if (!plot.second.empty()) {
           x_min = std::min(x_min, plot.second.front().first);
           x_max = std::max(x_max, plot.second.back().first);
         }
+      }
 
-      axesSetup(x_max);
+      axesSetup(x_min, x_max);
```
Comment (R-Abbasi) [inline src/main.cpp:135]: Where is `x_min` used, please?

File: src/main.cpp [hunk 5]
```diff
@@ -204,27 +208,41 @@ int main() {
...
     if (sampler.has_value())
       if (!flagPaused)
         sampler->copyOut(captureData);
 
-    ImGui::SetCursorPos({windowSize.x * 0.65f, windowSize.y * 0.05f});
-    ImGui::BeginChild("Buttonstripe", {-1, toolBtnSize.y}, false,
-                      ImGuiWindowFlags_NoScrollbar);
+    float optimal_buttonstripe_height = toolBtnSize.y * 1.1;
+    if (toolBtnSize.y < (ImGui::GetTextLineHeightWithSpacing() * 1.1))
+      optimal_buttonstripe_height = ImGui::GetTextLineHeightWithSpacing() * 1.1;
```
Comment (R-Abbasi) [inline src/main.cpp:225]: What is the point of using _optimal_buttonstripe_height_, please? `ImGui::GetTextLineHeightWithSpacing()` is the distance in pixels between 2 consecutive lines of text. Not quite sure what two lines of text.

File: src/main.cpp [hunk 6]
```diff
@@ -204,27 +208,41 @@ int main() {
...
-    ImGui::SetCursorPos({windowSize.x * 0.18f, windowSize.y * 0.06f});
+    // ##############################
+
+    ImGui::SetCursorPos(ImVec2(ImGui::GetIO().DisplaySize.x * 0.19f,
+                               ImGui::GetIO().DisplaySize.y * 0.06f));
+
     ImGui::BeginChild("Live Capture",
-                      {windowSize.x * 0.9f, windowSize.y * 0.68f});
+                      ImVec2(ImGui::GetIO().DisplaySize.x * 0.80f,
+                             ImGui::GetIO().DisplaySize.y * 0.65f));
```
Comment (R-Abbasi) [inline src/main.cpp:218]: By deleting `windowSize` code lines are going to increase reducing the readability and be less performant.

File: src/main.cpp [hunk 7]
```diff
@@ -373,8 +398,9 @@ int main() {
       ImGui::Button(appLanguage["Crt_trng_data"], toolBtnSize);
       ImGui::PopStyleColor();*/
     }
-    ImGui::EndChild(); // child "Buttonstripe"
```
Comment (R-Abbasi) [inline src/main.cpp:376]: Why did you delete that comment, please?

File: src/menuscpp/Style.cpp [hunk 8]
```diff
@@ -115,19 +116,35 @@ inline void SetupImGuiStyle(bool bStyleDark_, float alpha_) {
...
+
   ImGuiStyle &style = ImGui::GetStyle();
 
-  style.Colors[ImGuiCol_Text] = { 0.0f, 0.0f, 0.0f, 1.0f };
-  style.Colors[ImGuiCol_WindowBg] = { 1.0f, 1.0f, 0.93f, 1.0f };
-  style.Colors[ImGuiCol_Border] = { 0.14f, 0.15f, 0.17f, 1.0f };
-  style.Colors[ImGuiCol_FrameBg] = { 1.0f, 1.0f, 1.0f, 1.0f };
-  style.Colors[ImGuiCol_FrameBgHovered] = { 1.0f, 1.0f, 1.0f, 1.0f };
-  style.Colors[ImGuiCol_FrameBgActive] = { 1.0f, 1.0f, 1.0f, 1.0f };
+  style.Colors[ImGuiCol_Text] =
```
Comment (R-Abbasi) [inline src/menuscpp/Style.cpp:125]: Why did you change `{ 0.0f, 0.0f, 0.0f, 1.0f }` to `ImVec4(1 / 255.0f, 1 / 255.0f, 1 / 255.0f, 100 / 100.0f)`, please?

**Comment by DanielNowak98** [comment]
#152
